apps/tangle-dapp/src/components/account/ClaimRewardAction.tsx

```diff
@@ -0,0 +1,124 @@
+import { BN_ZERO } from '@polkadot/util';
+import { useActiveChain } from '@webb-tools/api-provider-environment/hooks/useActiveChain';
+import { TYPE_REGISTRY } from '@webb-tools/dapp-config/constants/polkadot';
+import Spinner from '@webb-tools/icons/Spinner';
+import { IconBase } from '@webb-tools/icons/types';
+import VipDiamondLine from '@webb-tools/icons/VipDiamondLine';
+import useTransactionInfo from '@webb-tools/tangle-shared-ui/hooks/useTransactionInfo';
+import { TangleAssetId } from '@webb-tools/tangle-shared-ui/types';
+import { useCallback, useEffect, useMemo } from 'react';
+import { formatUnits, zeroAddress } from 'viem';
+import useAccountRewardInfo from '../../data/rewards/useAccountRewardInfo';
+import useRewardsTx from '../../data/rewards/useRewardsTx';
+import useAgnosticAccountInfo from '../../hooks/useAgnosticAccountInfo';
+import ActionItem from './ActionItem';
+import { useWebbUI } from '@webb-tools/webb-ui-components/hooks/useWebbUI';
+import { twMerge } from 'tailwind-merge';
+
+export default function ClaimRewardAction() {
+  const { notificationApi } = useWebbUI();
+  const [activeChain] = useActiveChain();
+  const { isEvm } = useAgnosticAccountInfo();
+
+  const { claimRewards } = useRewardsTx();
+  const { isLoading, error, eventHandlers } = useTransactionInfo();
+
+  // TODO: Add other assets in the future,
+  // for now, we only support native asset
+  const claimableAssets = useMemo<TangleAssetId[]>(() => {
```

Comment:
Need to actually iterate over all assets the user has non-zero deposits in in vaults and then batch the calls / chunk them so that we can make efficient tx processing here.